docs/en/integrations/build-integration.asciidoc

```diff
@@ -306,20 +374,58 @@ apache
 │   │   └───elasticsearch/ingest_pipeline
 │   │   │      default.yml
 │   │   └───fields
-│   │          agent.yml <1>
-│   │          base-fields.yml <2>
-│   │          ecs.yml <3>
-│   │          fields.yml <4>
+│   │          agent.yml
+│   │          base-fields.yml
+│   │          ecs.yml
+│   │          fields.yml
 │   └───error
+│   │   └───elasticsearch/ingest_pipeline
+│   │   │      default.yml
+│   │   └───fields
+│   │          agent.yml
+│   │          base-fields.yml
+│   │          ecs.yml
+│   │          fields.yml
 │   └───status
 ----
-<1> `agent.yml` fields the Elastic agent uses
-<2> `base-fields.yml` never changes and is required for all integrations
-<3> Defines the relevant ECS fields
-<4> Custom Apache access log fields
 
+=== agent.yml
+The `agent.yml` file defines fields used by default processors.
+Examples: `cloud.account.id`, `container.id`, `input.type`
+
+=== base-fields.yml
+In this file, the `data_stream` subfields `type`, `dataset` and `namespace` are defined as type `constant_keyword`, the values for these fields are added by the integration.
+The `event.module` and `event.dataset` fields are defined with a fixed value specific for this integration:
+- `event.module: apache`
+- `event.dataset: apache.access`
+Field `@timestamp` is defined here as type `date`.
 
-// Need more on mapping
+=== ecs.yml
+This file specifies every Elastic Common Schema (ECS) field used by the integration that is not defined in the files `agent.yml` or `base-fields.yml` files. It uses `external: ecs` references.
+For example:
++
+[source,yaml]
+----
+- external: ecs
+  name: client.ip
+- external: ecs
+  name: destination.domain
+----
+
+=== fields.yml
```

Comment:
Can we provide a "Learn more" link to  https://github.com/elastic/integrations/blob/main/docs/generic_guidelines.md#document-all-field ?

Reply:
As of https://github.com/elastic/observability-docs/pull/3836, this content has moved to the integrations developer guide!

Reply:
Here's a link: https://www.elastic.co/guide/en/integrations-developer/current/general-guidelines.html#_document_all_fields